evm/its.js

```diff
@@ -437,6 +466,61 @@ async function processCommand(_, chain, options) {
             break;
         }
 
+        case 'checks': {
+            const interchainTokenService = new Contract(interchainTokenServiceAddress, InterchainTokenService.abi, wallet);
+
+            const contractConfig = chain.contracts[contractName];
+
+            const interchainTokenDeployer = await interchainTokenService.interchainTokenDeployer();
+            const interchainTokenFactory = await interchainTokenService.interchainTokenFactory();
+
+            const interchainTokenFactoryContract = new Contract(interchainTokenFactory, InterchainTokenFactory.abi, wallet);
+            const interchainTokenFactoryImplementation = await interchainTokenFactoryContract.implementation();
+
+            const interchainTokenDeployerContract = new Contract(interchainTokenDeployer, IInterchainTokenDeployer.abi, wallet);
+            const interchainToken = await interchainTokenDeployerContract.implementationAddress();
+
+            const [trustedChains, trustedAddresses] = await getTrustedChainsAndAddresses(config, interchainTokenService);
+
+            printInfo('Trusted chains', trustedChains);
+            printInfo('Trusted addresses', trustedAddresses);
+
+            const gateway = await interchainTokenService.gateway();
+            const gasService = await interchainTokenService.gasService();
+
+            const configGateway = chain.contracts.AxelarGateway?.address;
+            const configGasService = chain.contracts.AxelarGasService?.address;
+
+            const chainNameHash = await interchainTokenService.chainNameHash();
+            const configChainNameHash = keccak256(toUtf8Bytes(chain.id));
+
+            compare(gateway, configGateway, 'AxelarGateway');
+            compare(gasService, configGasService, 'AxelarGasService');
+            compare(chainNameHash, configChainNameHash, 'chainNameHash');
+
+            const toCheck = {
+                tokenManagerDeployer: await interchainTokenService.tokenManagerDeployer(),
+                interchainTokenDeployer,
+                interchainToken,
+                tokenManager: await interchainTokenService.tokenManager(),
+                tokenHandler: await interchainTokenService.tokenHandler(),
+                implementation: await interchainTokenService.implementation(),
+                interchainTokenFactory,
+                interchainTokenFactoryImplementation,
+            };
+
+            for (const [key, value] of Object.entries(toCheck)) {
+                if (Object.prototype.hasOwnProperty.call(contractConfig, key)) {
```

Comment:
```suggestion
                if (contractConfig[key]) {
```
isn't this sufficient?